CI: Trigger tag on CI changes

dsotirakis opened the pull request:
**What this PR does / why we need it**:

Triggers tag build on CI changes. Every time `.drone.yml` or `pkg/build/**` files are touched, along with the PR build, we are going to have a `v0.0.0-test` tag which will get triggered from this very PR. 

**Context**:
Unfortunately there's no way to run 2 builds in the same PR on Drone. This makes testing various conditional functionalities difficult. What we need 2 builds for, is to be able to run our PR pipelines normally, and upon CI related changes, run our release/publishing process so we can make sure that we don't break anything.
Currently we do this check after we merge to main, which works but it would be better to have the feedback in the same PR.

**Caveat**:
When this PR is in sync with the latest `origin main` branch, we get both `continuous-integration/drone/pr` and `continuous-integration/drone/tag` running in the PR checks.
Example:
![Screenshot 2022-10-07 at 10 29 29 AM](https://user-images.githubusercontent.com/15115078/194497355-d6225ce2-c792-49a2-ad49-788547c7fba9.png)
And then when the tag build runs, that's how it looks in Drone:
![image](https://user-images.githubusercontent.com/15115078/194497403-ed7aab52-87a1-45aa-a5e7-28fd808ddc0a.png)

-------

If the PR is not in sync with main, what we get is the normal PR running, and a separate build on Drone, which doesn't appear in the PR checks.
Example:
![Screenshot 2022-10-07 at 10 26 51 AM](https://user-images.githubusercontent.com/15115078/194496907-9a68207f-a2cf-4cd5-b865-6e8db416590e.png)
And then when the tag build runs, that's how it looks in Drone:
![image](https://user-images.githubusercontent.com/15115078/194497132-01fc661c-85a9-49ff-a2f1-3f623eaad69d.png)

malcolmholmes commented on scripts/drone/events/release.star:
```diff
@@ -462,9 +462,6 @@ def release_pipelines(ver_mode='release', trigger=None):
             'ref': ['refs/tags/v*',],
-            'repo': {
-                'exclude': ['grafana/grafana'],
-            },
```
Hmmm. This is probably okay. Habit is sufficiently in place to not make this a problem.

dsotirakis replied:
Yes! No problem with that, upon tagging, release builds are gonna run and put things into buckets. All good.

dsotirakis committed:
Trigger tag on CI changes

dsotirakis committed:
Use github token available for PRs

Remove remote addition

dsotirakis committed:
Replace origin with full url

Remove fetch for origin

Fix fetch

Remove --quite

Add rebase

Stash changes from verify-gen-cue

Cleanup

dsotirakis committed:
Add --quiet when fetching
